assets/pages/new-contributions-landing/components/ContributionForm.jsx

```diff
@@ -35,236 +29,159 @@ import { NewContributionSubmit } from './ContributionSubmit';
-          },
-        };
+function onSubmit(props: PropTypes): Event => void {
+  return (event) => {
+    event.preventDefault();
+    const amount = getAmount(props);
+    const { email } = props;
+
+    if (props.paymentHandler) {
+      switch (props.paymentMethod) {
+        case 'DebitCard':
+          // TODO
+          break;
+
+        case 'PayPal':
+          // TODO
+          break;
+
+        case 'Stripe':
+        default:
+          if (props.paymentHandler.Stripe) {
+            openDialogBox(props.paymentHandler.Stripe, amount, email);
+          }
+          break;
+      }
     }
   };
 }
 
-// ----- Event handlers ----- //
-
-const onSubmit = form => (stripeHandler) => {
-  const { elements } = (form: any);
-  const amount = getAmount(elements);
-  const email = elements.namedItem('contributionEmail').value;
-
-  openDialogBox(stripeHandler, amount, email);
-};
-
-
 // ----- Render ----- //
 
-function setupStripe(formElement: Object, props: PropTypes) {
-  const {
-    csrf,
-    currency,
-    contributionType,
-    abParticipations,
-    isTestUser,
-  } = props;
-
-  const callback: PaymentCallback = bracketPromise(
-    () => { props.onWaiting(true); return Promise.resolve(); },
-    () => { props.onWaiting(false); return Promise.resolve(); },
-    createPaymentCallback(
-      getData(props, formElement),
-      contributionType,
-      abParticipations,
-      csrf,
-    ),
-  );
-
-  const onSuccess: PaymentResult => void = (result) => {
-    switch (result.paymentStatus) {
-      case 'success':
-        trackConversion(abParticipations, '/contribute/thankyou.new');
-        props.onSuccess();
-        break;
-
-      default:
-        props.onError(result.error);
-    }
-  };
-
-  return setupStripeCheckout(callback, contributionType, currency, isTestUser, onSuccess);
-}
-
 function ContributionForm(props: PropTypes) {
   const {
     countryGroupId,
     selectedCountryGroupDetails,
-    currency,
     thankYouRoute,
-    initialFirstName,
-    initialLastName,
-    initialEmail,
+    firstName,
+    lastName,
+    email,
   } = props;
 
+  const paymentCallback = (token: Token) => {
+    props.onWaiting(true);
+    props.onThirdPartyPaymentDone(token);
+  };
+
   return props.done ?
     <Redirect to={thankYouRoute} />
     : (
       <div className="gu-content__content">
         <h1>{countryGroupSpecificDetails[countryGroupId].headerCopy}</h1>
         <p className="blurb">{countryGroupSpecificDetails[countryGroupId].contributeCopy}</p>
-        <ErrorMessage message={props.error} />
-        <form
-          onSubmit={(e) => {
-            e.preventDefault();
-            const formElement = e.target;
-            if (formElement) {
-              setupStripe(formElement, props).then(onSubmit(formElement));
-            }
-        }}
-          className={classNameWithModifiers('form', ['contribution'])}
-        >
+        <PaymentFailureMessage checkoutFailureReason={props.error} />
```

Comment:
@Amohkhan there is no style for failures during the payment process, can you quickly do something for that?